signer/src/transaction_signer.rs

```diff
@@ -140,6 +141,9 @@ pub struct TxSignerEventLoop<Context, Network, Rng> {
     pub wsts_state_machines: LruCache<StateMachineId, SignerStateMachine>,
     /// The threshold for the signer
     pub threshold: u32,
+    /// Set of bitcoin blocks for which the signer has already processed
+    /// presign request.
+    pub used_presign_blocks: HashSet<BitcoinBlockRef>,
```

Comment:
I think we can simplify this and just keep the latest chain tip hash for which we signed a pre sign: we enter in `handle_bitcoin_pre_sign_request` only if the message is for a `ChainTipStatus::Canonical` chain tip and sent by the coordinator, so if we just keep track of the latest chain tip we signed for and bail if the new request matches it it should be enough.

Reply:
[87b0059](https://github.com/stacks-sbtc/sbtc/pull/1757/commits/87b005954edf255d90eeee1e8588c27ed76d80ac)